affiliate/webpack.config.js

```diff
@@ -0,0 +1,67 @@
+// const defaultConfig = require('@wordpress/scripts/config/webpack.config');
+//
+// module.exports = [
+//   {
+//     ...defaultConfig,
+//     entry: './src/productCard/index.js',
+//     output: './build/productCard/',
+//   },
+//   {
+//     ...defaultConfig,
+//     entry: './src/productCarousel/index.js',
+//     output: './build/productCarousel/',
+//   },
+// ];
+
+/**
+ * `@wordpress/scripts` path-based name multi-block Webpack configuration.
+ * @see https://wordpress.stackexchange.com/questions/390282
+ */
+
+const config = require('@wordpress/scripts/config/webpack.config.js');
+const CopyPlugin = require('copy-webpack-plugin');
+const path = require('path');
+
+/**
+ * Resolve a series of path parts relative to `./src`.
+ * @param string[] pathParts An array of path parts.
+ * @returns string A normalized path, relative to `./src`.
+ * */
+const resolveSource = (...pathParts) => path.resolve(process.cwd(), 'src', ...pathParts);
+
+/**
+ * Resolve a block name to the path to it's main `index.js` entry-point.
+ * @param string name The name of the block.
+ * @returns string A normalized path to the block's entry-point file.
+ * */
+const resolveBlockEntry = (name) => resolveSource(name, 'index.js');
+
+config.entry = {
+  'productCard/index': resolveBlockEntry('productCard'),
+  'productCarousel/index': resolveBlockEntry('productCarousel'),
+};
```

Comment:
I've been trying to get this directory to accept multiple blocks through a few different methods and this is the closest I have been able to get to that goal. At first, I was hoping it would be as simple as [this Github thread](https://github.com/WordPress/gutenberg/issues/40771#issuecomment-1127772731) makes it out to be rather simple but the build directory would be out of sorts whenever I tried it, so I started trying to change the Webpack build process. Which led me to [this stackexchange post](https://wordpress.stackexchange.com/questions/390282) but it is still not working as hoped.

Currently, the Product Carousel seems to be recognized by WP, can be placed on the page, and I see the modal working (haven't been able to fully test because it still uses the production version of the product carousel creation page and I only just now pushed the [PR](https://github.com/orgnc/platform/pull/2745)). However, the Product Card isn't showing up in WP. I also dislike using this Webpack config file, since it is pretty messy and would like to be able to use the method in that first Github link, but it has been a problem.

Reply:
I wonder what you think about this sort of refactor? https://github.com/orgnc/wordpress-plugin/pull/131. (Definitely needs cleaning up! For instance, if we did things this way, we'd need to fix the build-zip.sh.) I've gotten things to work with a single IntegrationModal, and changing package.json to run differently. Still getting the spacing weirdness though.

Reply:
So instead of having one src directory, we have a blocks directory; inside, for each integration type, we have a directory with the block and src. Our package.json build script builds each of these separately.

Reply:
In any case, this refactor gets rid of the config.

Reply:
Man, I used that same sort of `wp-scripts build` line with the output-path but it didn't seem to work... Is it just because of the nested `blocks` directory? Or, I might just be at constant odds with this dev environment. Personally, I don't like the deeply nested structure and would like to keep the directory structure similar to this PR since these blocks aren't supposed to have a lot of moving parts so adding more nested directories ends up looking more confusing.

But, I'm good with that direction if the build script doesn't end up working with this PR's directory setup.

Reply:
Also, I think this whitespace might be a problem with the dev environment. I'm not seeing what could have caused this and it appears even when I switch back to the master branch...

Reply:
> Or, I might just be at constant odds with this dev environment.

I spent, like, another hour getting it working again, so it's not just you!

I _think_ the current directory structure should work. I think the following should do the trick:

- Be sure that there are two separate build directories (otherwise I believe one will overwrite the other), so, e.g., src/productCard/build and src/productCarousel/build. The build script (which I think we should rename to "affiliate-integrations") should run wp-scripts build <productCard> && wp-scripts build <productCarousel>.
- Make sure to repoint the register-scripts stuff in Affiliate.php to the new locations.

Reply:
> Also, I think this whitespace might be a problem with the dev environment. I'm not seeing what could have caused this and it appears even when I switch back to the master branch...

Wondering this myself. I'll post to dev-environment.